src/ray/common/cgroup/cgroup_setup.cc

```
@@ -87,37 +90,23 @@ namespace internal {
 
 bool CanCurrenUserWriteCgroupV2() { return access("/sys/fs/cgroup", W_OK | X_OK) == 0; }
 
-bool IsCgroupV2MountedAsRw() {
-  // Checking all mountpoints directly and parse textually is the easiest way, compared
-  // with mounted filesystem attributes.
-  std::ifstream mounts("/proc/mounts");
-  if (!mounts.is_open()) {
+bool IsCgroupV2MountedAsRw(const std::string &path) {
+  // Check whether cgroupv2 is mounted solely at `/sys/fs/cgroup`.
+  struct statfs fs_stats;
+  if (statfs(path.data(), &fs_stats) != 0) {
     return false;
   }
-
-  // Mount information is formatted as:
-  // <fs_spec> <fs_file> <fs_vfstype> <fs_mntopts> <dump-field> <fsck-field>
-  std::string line;
-  while (std::getline(mounts, line)) {
-    std::vector<std::string_view> mount_info_tokens = absl::StrSplit(line, ' ');
-    RAY_CHECK_EQ(mount_info_tokens.size(), 6UL);
-    // For cgroupv2, `fs_spec` should be `cgroupv2` and there should be only one mount
-    // information item.
-    if (mount_info_tokens[0] != "cgroup2") {
-      continue;
-    }
-    const auto &fs_mntopts = mount_info_tokens[3];
-
-    // Mount options are formatted as: <opt1,opt2,...>.
-    std::vector<std::string_view> mount_opts = absl::StrSplit(fs_mntopts, ',');
-
-    // CgroupV2 has only one mount item, directly returns.
-    return std::any_of(mount_opts.begin(),
-                       mount_opts.end(),
-                       [](const std::string_view cur_opt) { return cur_opt == "rw"; });
+  if (fs_stats.f_type != CGROUP2_SUPER_MAGIC) {
+    return false;
```

These are all FATAL errors now. Does it still make sense to return false?

I'm thinking failing a CHECK and logging the error type closer to where it happens would be more useful for debugging failures

---

The fail fast pattern will be exercised here: https://github.com/ray-project/ray/pull/49941

For this PR, returning true/false is enough, and it should be merged into main before the above one.

---

For testing purposes it's probably better to return true/false from this function and add the check/fatal error at the callsite.

Would recommend returning a user-facing string describing the aspect of this function that failed. For example, `"cgroup v2 mount at <path> is not writeable"`